datatypes/src/primitives/spatial_partition.rs

```diff
@@ -83,9 +83,9 @@ impl SpatialPartition2D {
     /// Checks if a coordinate is located inside spatial partition
     pub fn contains_coordinate(&self, coordinate: &Coordinate2D) -> bool {
         coordinate.x >= self.upper_left_coordinate.x
-            && coordinate.y >= self.upper_left_coordinate.y
+            && coordinate.y <= self.upper_left_coordinate.y
             && coordinate.x < self.lower_right_coordinate.x
-            && coordinate.y < self.lower_right_coordinate.y
+            && coordinate.y > self.lower_right_coordinate.y
```

Comment:
Why does the logic change when changing into a new `proj` version?
Was this a bug? If so, we should write a test for that.